sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScalerProvider.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+using Microsoft.Azure.WebJobs.Extensions.ServiceBus.Config;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.ServiceBus;
+using Microsoft.Azure.WebJobs.ServiceBus.Listeners;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ServiceBus.Listeners
+{
+    internal class ServiceBusScalerProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly IConfiguration _configuration;
+        private AzureComponentFactory _defaultAzureComponentFactory;
+        private readonly MessagingProvider _messagingProvider;
+        private readonly IOptions<ServiceBusOptions> _options;
+        private readonly AzureEventSourceLogForwarder _logForwarder;
+        private readonly TriggerMetadata _triggerMetadata;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly ServiceBusMetadata _serviceBusMetadata;
+        private readonly AzureComponentFactory _azureComponentFactory;
+
+        public ServiceBusScalerProvider(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            _configuration = serviceProvider.GetService<IConfiguration>();
+            _defaultAzureComponentFactory = triggerMetadata.Properties[nameof(AzureComponentFactory)] as AzureComponentFactory;
```

Comment:
Here and below where we're using values from the user supplied triggerMetadata the various properties can be null I think. I don't think TriggerMetadata does null checks in its ctors - it should.

Reply:
Also, if the factory isn't in the properties, won't this throw?